tokio/src/task/local.rs

```diff
@@ -667,21 +674,27 @@ impl LocalSet {
     }
 
     fn with<T>(&self, f: impl FnOnce() -> T) -> T {
-        CURRENT.with(|LocalData { ctx, .. }| {
+        CURRENT.with(|LocalData { ctx, entered, .. }| {
             struct Reset<'a> {
                 ctx_ref: &'a RcCell<Context>,
-                val: Option<Rc<Context>>,
+                entered_ref: &'a Cell<bool>,
```

Comment:
nit, not a big deal: both `ctx_ref` and `entered_ref` reference fields of the `LocalData`. we could make this struct a word smaller by changing these to a single `&'a LocalData` reference, and just using that to access both `ctx` and `entered`.

Reply:
It's a great idea! I've moved out, and renamed to more verbose name.